engine-tests/src/test/java/org/terasology/config/flexible/validators/RangedNumberValidatorTest.java

```diff
@@ -0,0 +1,210 @@
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
+
+            assertFalse(validator.fastValidate(validator.getMin()));
+            assertTrue(validator.fastValidate(validator.getMax()));
+        }
+
+        @Test
+        public void testMaxExclusive() {
+            initValidator(0, 100, true, false);
+
+            for (int i = validator.getMin() + 1; i < validator.getMax(); i++) {
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
+
+            assertTrue(validator.fastValidate(validator.getMin()));
+            assertFalse(validator.fastValidate(validator.getMax()));
+        }
+
+        @Test
+        public void testAllExclusive() {
+            initValidator(0, 100, false, false);
+
+            for (int i = validator.getMin() + 1; i < validator.getMax(); i++) {
+                assertTrue(String.format("%d returned invalid", i), validator.fastValidate(i));
+            }
+
+            assertFalse(validator.fastValidate(validator.getMin()));
+            assertFalse(validator.fastValidate(validator.getMax()));
+        }
+
+        @Test
+        public void testLowUnbounded() {
+            initValidator(null, 100, false, false);
+
+            assertTrue(validator.fastValidate(-1000));
+            assertTrue(validator.fastValidate(-50000));
+
+            assertTrue(validator.fastValidate(50));
+
+            assertFalse(validator.fastValidate(validator.getMax() + 1));
+        }
+
+        @Test
+        public void testHighUnbounded() {
+            initValidator(0, null, false, false);
+
+            assertTrue(validator.fastValidate(1000));
+            assertTrue(validator.fastValidate(50000));
+
+            assertTrue(validator.fastValidate(50));
+
+            assertFalse(validator.fastValidate(validator.getMin() - 1));
+        }
+
+        @Test
+        public void testAllUnbounded() {
+            initValidator(null, null, false, false);
+
+            assertTrue(validator.fastValidate(1000));
+            assertTrue(validator.fastValidate(50000));
+            assertTrue(validator.fastValidate(50));
+
+            assertTrue(validator.fastValidate(-1000));
+            assertTrue(validator.fastValidate(-50000));
+        }
+    }
+
+    public static class DoubleValidator {
+        private static final double EPSILON = 0.000001d;
+
+        private RangedNumberValidator<Double> validator;
+
+        private void initValidator(Double min, Double max, boolean minInclusive, boolean maxInclusive) {
+            validator = new RangedNumberValidator<>(min, max, minInclusive, maxInclusive);
+        }
+
+        @Test
+        public void testAllInclusive() {
+            initValidator(0d, 100d, true, true);
+
+            for (double i = validator.getMin(); i <= validator.getMax(); i++) {
+                assertTrue(String.format("%f returned invalid", i), validator.fastValidate(i));
+            }
```

Comment:
Same comment as above: use a much smaller range and a few validation lines rather than a loop please.